zip-0312.rst

```diff
@@ -1,7 +1,451 @@
+- The transaction is signed with a re-randomized version of the user's spend
+  authorization private key.
+- The zero-knowledge proof for the transaction is created with the randomizer
+  as an auxiliary (secret) input, among others.
+
+When employing re-randomizable FROST as specified in this ZIP, the goal is to
+split the spend authorization private key :math:`\mathsf{ask}` among multiple
+possible signers. This means that the proof generation will still be performed
+by a single participant, likely the one that created the transaction in the first
+place. Note that this user already controls the privacy of the transaction since
+they are responsible for creating the proof.
+
+This fits well into the "Coordinator" role from the FROST specification
+[#frost-protocol]_. The Coordinator is responsible for sending the message to be
+signed to all participants, and to aggregate the signature shares.
+
+With those considerations in mind, the threat model considered in this ZIP is:
+
+- The Coordinator is trusted with the privacy of the transaction (which includes
+  the unlinkability property). A rogue Coordinator will be able to break
+  unlinkability and privacy, but should not be able to create signed transactions
+  without the approval of ``MIN_PARTICIPANTS`` participants, as specified in FROST.
+- All key share holders are also trusted with the privacy of the transaction,
+  thus a rogue key share holder will be able to break its privacy and unlinkability.
+
+
+Non-requirements
+================
+
+- This ZIP does not support removing the Coordinator role, as described in
+  [#frost-removingcoordinator]_.
+- This ZIP does not prevent key share holders from linking the signing operation to a
+  transaction in the blockchain.
+- Like the FROST specification [#FROST]_, this ZIP does not specify a key generation
+  procedure; but refer to that specification for guidelines.
+- Network privacy is not in scope for this ZIP, and must be obtained with other
+  tools if desired.
+
+
+Specification
+=============
+
+Algorithms in this section are specified using Python pseudo-code, in the same
+fashion as the FROST specification [#FROST]_.
+
+The types Scalar, Element, and G are defined in [#frost-primeordergroup]_, as
+well as the notation for elliptic-curve arithmetic, which uses the additive
+notation. Note that this notation differs from that used in the Zcash Protocol
+Specification. For example, ``G.ScalarMult(P, k)`` is used for scalar
+multiplication, where the protocol spec would use :math:`[k] P` with the group
+implied by :math:`P`.
+
+An additional per-ciphersuite hash function is used, denote ``HR(m)``, which
+receives an arbitrary-sized byte string and returns a Scalar. It is defined
+concretely in the Ciphersuites section.
+
+
+Re-randomizable FROST
+---------------------
+
+To add re-randomization to FROST, follow the specification [#FROST]_ with the
+following modifications.
+
+
+Key Generation
+''''''''''''''
```

Comment:
Move this subsection to instead be a section before "Re-randomizable FROST". Given that FROST does not specify key generation, it doesn't make sense to define this section as a modification to the FROST specification.

Reply:
Done in 6e6893d8323103013163586eb7e3c53e929a14b4